Suggestions to make model easier to run

cc-a opened the pull request:
Hi. Gave the model a quick spin to see if I could get it working. This pull request implements some suggestions to make getting up and running with the model quicker and easier for other people:

1. Explicitly list the required R packages for running the model. This should make it easier for people to install all the dependencies up front.
2. A Dockerfile and instructions on running the model with Docker. This provides a way for users to install all dependencies and run the model with just two commands.

Should you choose to merge these, as a colleague at Imperial I'm happy to field any queries relating to Docker and undertake any maintenance required. If you're interested in exploring this route further it is possible to [upload a pre-built Docker image to GitHub](https://github.com/features/packages) so that users could run the entire model with just a single command. Building and testing of the pre-built image could also be carried out using [GitHub actions](https://github.com/features/actions). This is all freely available under the Imperial GitHub organisation.

cc-a committed:
Added Dockerfile and instructions for running model with docker

cc-a committed:
Added details of R package dependencies to README

cc-a commented:
After sleeping on it, it occurs to me that an environment.yml file would be a good option as well to allow users to easily create a conda environment with all dependencies installed. Happy to knock one together if that seems like a good idea.

harrisonzhu508 commented:
Agree with the environment.yml - would be good to specify the minimum version of R as well for this to run. @s-mishra @cc-a

cc-a commented:
See prototype below. I'll wait for a steer on whether to include this as part of the PR.
```
name: covid19model
dependencies:
  - r-bayesplot
  - r-data.table
  - r-envstats
  - r-lubridate
  - r-gdata
  - r-ggpubr
  - r-rstan
channels:
  - conda-forge
```

flaxter commented:
This is great. @Ettie, we can remove dependency on ggpubr, right? Any
others that are easy to swap out?

On Thu, 2 Apr 2020, 11:16 Christopher Cave-Ayland, <notifications@github.com>
wrote:

> See prototype below. I'll wait for a steer on whether to include this as
> part of the PR.
>
> name: covid19model
> dependencies:
>   - r-bayesplot
>   - r-data.table
>   - r-envstats
>   - r-lubridate
>   - r-gdata
>   - r-ggpubr
>   - r-rstan
> channels:
>   - conda-forge
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ImperialCollegeLondon/covid19model/pull/4#issuecomment-607754466>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAOPW6IQEC2GKAG6LB2QM3RKRQXLANCNFSM4LZQNB4Q>
> .
>

s-mishra commented:
Hi Chris, 

Go ahead and make the environment.yml too. Thanks for the effort @cc-a. Once done I will accept the pull request. @harrisonzhu508  can you just cross-check the chnage?

harrisonzhu508 commented:
No problem, I'll test the container and let you know.

ettieunwin commented:
Yeah - ggpubr is just to make the plots look like the other Imperial ones but it's not necessary

cc-a committed:
Added environment.yml

cc-a committed:
Reformatted README with a dedicated section on installing dependencies

cc-a committed:
Removed accidental change to README formatting

dentarthur commented:
Hi @cc-a and @s-mishra 

Please check related suggestion I have placed here:

https://github.com/thecapitalistcycle/covid19model/issues/1

I have put it there to keep it out of the way and not create a separate issue for a suggestion I cannot actually contribute to.

cc-a commented:
Hi, I've added the `environment.yml` and added a dedicated section to the README on "installing dependencies".

If you do decide to remove ggpubr as a dependency this can then be removed from both `environment.yml`and `docker/Dockerfile`.

Please let me know if you'd like any further changes or are interested in automating some testing around these.

harrisonzhu508 commented:
Thanks a lot for the suggestions! Would be interested to explore the GitHub actions aspect or automated testing. Just building the docker image at the moment but will let you know if it's all good on my machine.

harrisonzhu508 commented:
@cc-a  @s-mishra  Ran the container and was all good from my side. I suggest we merge after we discuss some automatic testing or adding pre-built docker images into GitHub?

cc-a commented:
Just one thought before this is merged, instructions for running Docker may need to be different for Windows just doing a quick check.

cc-a committed:
Add to .gitattributes to ensure shell scripts have LF line endings

This is required to ensure correct line endings when building the Docker image on Windows.

cc-a committed:
Update README.md

cc-a commented:
Made a couple of minor tweaks to get the build and run to work with Docker on Windows.

cc-a committed:
Updated windows docker instructions

s-mishra commented:
Hi Chris,

Thanks for your efforts. Can I ask one thing? Do we have GitHub actions in this pull request so that running and reproducing is easy every time we push an update? If this is something you can do easily then please do and we will accept the PR. Thanks again for your work and time.

cc-a commented:
As is this does not include use of GitHub actions. With the amount of time I can realistically spend putting that together would take me a day or two I expect. It's a bit tricky to put together without admin access to the repo as well.

cc-a commented:
Maybe best if I raise a separate issue and suggest what I think is possible, the functionality it would provide and what would be needed to implement. Then we could figure out how to go forward from there?

harrisonzhu508 commented:
That sounds good, I think we can merge. @s-mishra @cc-a

flaxter commented:
Thanks Christopher, really appreciate it.

On Thu, 2 Apr 2020, 17:25 hbz, <notifications@github.com> wrote:

> That sounds good, I think we can merge. @s-mishra
> <https://github.com/s-mishra> @cc-a <https://github.com/cc-a>
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ImperialCollegeLondon/covid19model/pull/4#issuecomment-607949662>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAOPW3DLIUWHKBN2SO64STRKS35VANCNFSM4LZQNB4Q>
> .
>

bhattsamir commented:
Thank you, Christopher.

On Thu, 2 Apr 2020 at 17:45, Seth Flaxman <notifications@github.com> wrote:

> Thanks Christopher, really appreciate it.
>
> On Thu, 2 Apr 2020, 17:25 hbz, <notifications@github.com> wrote:
>
> > That sounds good, I think we can merge. @s-mishra
> > <https://github.com/s-mishra> @cc-a <https://github.com/cc-a>
> >
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/ImperialCollegeLondon/covid19model/pull/4#issuecomment-607949662
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AAAOPW3DLIUWHKBN2SO64STRKS35VANCNFSM4LZQNB4Q
> >
> > .
> >
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ImperialCollegeLondon/covid19model/pull/4#issuecomment-607960831>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACYTMIUAFETLQCZBOADV2ZTRKS6I7ANCNFSM4LZQNB4Q>
> .
>


-- 
*Dr Samir Bhatt*


*Senior Lecturer (Associate Professor) in GeostatisticsThe Department of
Infectious Disease EpidemiologyImperial College London*
<http://www.map.ox.ac.uk/> <http://www.map.ox.ac.uk/>

harrisonzhu508 commented:
Thank you Christopher, going to merge it now.
